pkg/kbagent/util/event.go

```diff
@@ -21,56 +21,43 @@ package util
 
 import (
 	"context"
-	"encoding/json"
-	"errors"
 	"fmt"
 	"os"
 	"time"
 
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
 	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/rand"
+	"k8s.io/client-go/kubernetes"
 	ctlruntime "sigs.k8s.io/controller-runtime"
 
 	"github.com/apecloud/kubeblocks/pkg/constant"
 )
 
-var logger = ctlruntime.Log.WithName("event")
-var EventSendMaxAttempts = 30
-var EventSendPeriod = 10 * time.Second
-
-func SentEventForProbe(ctx context.Context, msg ActionMessage) error {
-	logger.Info(fmt.Sprintf("send event: %v", msg))
-	action := msg.GetAction()
-	if action == "" {
-		return errors.New("action is unset")
-	}
-	event, err := CreateEvent(action, msg)
-	if err != nil {
-		logger.Info("create event failed", "error", err.Error())
-		return err
-	}
+const (
+	sendEventMaxAttempts   = 30
+	sendEventRetryInterval = 10 * time.Second
+)
 
+func SendEventWithMessage(logger *logr.Logger, reason string, message string) {
 	go func() {
-		_ = SendEvent(ctx, event)
+		event := createEvent(reason, message)
+		err := sendEvent(event)
+		if logger != nil && err != nil {
+			logger.Error(err, "send event failed")
+		}
 	}()
-
-	return nil
 }
 
-func CreateEvent(reason string, msg ActionMessage) (*corev1.Event, error) {
-	// get pod object
+func createEvent(reason string, message string) *corev1.Event {
 	podName := os.Getenv(constant.KBEnvPodName)
```

Comment:
If there are no KB_POD_XXX environment variables, this may not function correctly.

Reply:
Add a TODO to tracking it.